pkgs/development/haskell-modules/configuration-ghcjs-9.x.nix

```diff
@@ -0,0 +1,47 @@
+{ pkgs, haskellLib }:
+
+with haskellLib;
+
+pkgs.lib.composeManyExtensions [
+
+  (self: super: {
+    ghcjs-base = super.ghcjs-base_0_8_0_0;
+  })
+
+  (self: super: {
+    ghcjs-dom-javascript = unmarkBroken super.ghcjs-dom-javascript;
+    ghcjs-dom-jsffi = unmarkBroken super.ghcjs-dom-jsffi;
```

Comment:
I think I shouldn't need these, but `cabal2nix` gets [confused](https://github.com/NixOS/cabal2nix/blob/f8e6bf749a158a5ed866c57deee907b5f16c38e5/distribution-nixpkgs/CHANGELOG.md#unreleased) when I try to set the supported platform. When given "javascript-ghcjs" it outputs "js-ghcjs" which then leads to
```
 error: Package ‘ghcjs-dom-javascript-0.9.9.0’ in /home/alexfmpe/profiles/alexfmpe/repos/nixpkgs/pkgs/development/haskell-modules/hackage-packages.nix:120114 is not available on the requested hostPlatform:
         hostPlatform.config = "javascript-unknown-ghcjs"
         package.meta.platforms = [
           "js-ghcjs"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.
```

This is what I tried, including generated `hackage-packages.nix` : 

```diff
diff --git a/pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml b/pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml
index 66895c194519..e44096f7a52f 100644
--- a/pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml
+++ b/pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml
@@ -1989,8 +1989,6 @@ broken-packages:
   - ghci-ng # failure in job https://hydra.nixos.org/build/233229533 at 2023-09-02
   - ghcitui # failure in job https://hydra.nixos.org/build/252737339 at 2024-03-16
   - ghcjs-base-stub # timeout
-  - ghcjs-dom-javascript # failure in job https://hydra.nixos.org/build/255688382 at 2024-04-16
-  - ghcjs-dom-jsffi # failure in job https://hydra.nixos.org/build/233215225 at 2023-09-02
   - ghcjs-fetch # timeout
   - ghcjs-promise # failure in job https://hydra.nixos.org/build/233243985 at 2023-09-02
   - ghcjs-xhr # failure in job https://hydra.nixos.org/build/233235693 at 2023-09-02
diff --git a/pkgs/development/haskell-modules/configuration-hackage2nix/main.yaml b/pkgs/development/haskell-modules/configuration-hackage2nix/main.yaml
index 0f0b3520b2e0..125a0774d3fa 100644
--- a/pkgs/development/haskell-modules/configuration-hackage2nix/main.yaml
+++ b/pkgs/development/haskell-modules/configuration-hackage2nix/main.yaml
@@ -721,6 +721,8 @@ supported-platforms:
   geomancy:                                     [ platforms.x86 ] # x86 intrinsics
   geomancy-layout:                              [ platforms.x86 ] # x86 intrinsics
   gi-gtkosxapplication:                         [ platforms.darwin ]
+  ghcjs-dom-javascript:                         [ javascript-ghcjs ]
+  ghcjs-dom-jsffi:                              [ javascript-ghcjs ]
   gtk-mac-integration:                          [ platforms.darwin ]
   gtk3-mac-integration:                         [ platforms.darwin ]
   halide-haskell:                               [ platforms.linux ]
diff --git a/pkgs/development/haskell-modules/hackage-packages.nix b/pkgs/development/haskell-modules/hackage-packages.nix
index 4ceb1c7be6ae..3952740cfa54 100644
--- a/pkgs/development/haskell-modules/hackage-packages.nix
+++ b/pkgs/development/haskell-modules/hackage-packages.nix
@@ -120116,8 +120116,7 @@ self: {
        sha256 = "0vhk1gjr5g5cp6rzkr9p0km4mrdi61kfwk8jc0w97wkwmi030qcb";
        description = "DOM library using JSFFI and GHCJS";
        license = lib.licenses.mit;
-       hydraPlatforms = lib.platforms.none;
-       broken = true;
+       platforms = [ "js-ghcjs" ];
      }) {};
 
   "ghcjs-dom-jsaddle" = callPackage
@@ -120140,8 +120139,7 @@ self: {
        sha256 = "0fhcs89x180kw75qgbfh28wjyn55frwdfj4rqlqa1smx0fwhzyy1";
        description = "DOM library using JSFFI and GHCJS";
        license = lib.licenses.mit;
-       hydraPlatforms = lib.platforms.none;
-       broken = true;
+       platforms = [ "js-ghcjs" ];
      }) {};
 
   "ghcjs-dom-webkit" = callPackage
```

Reply:
This “just” needs a fix in distribution-nixpkgs. The relevant code there was written before GHC upstream/Cabal forced us to adopt the javascript arch prefix and I forgot to change it there, too.

Reply:
Fixed by the newly merged release of distribution-nixpkgs